### PR TITLE
Add New Account Modal Edits

### PR DESCRIPTION
**Pull request by rayykim**
Increased height of select box.
Bolded text in each field.
Updated modal width for smaller window.

**Before updated**
screen shot 2016-08-11 at 10 20 47 am

**Updated Modal**
screen shot 2016-08-11 at 10 43 41 am

*Short description*

## TODO

- [ ] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

List the issues solved or partly solved by the PR

## Documentation

If the PR has documentation, link the file here (either .rst in your repo or if built on Read The Docs)

## Screenshots (if appropriate)

They're nice. :)

### CODE REVIEW

File: kolibri/plugins/management/assets/src/vue/modal/index.vue [hunk 1]
```diff
@@ -85,8 +86,14 @@
...
   .modal-container
     background: #fff
+<<<<<<< HEAD
+    width: 85%
+    height: auto
+    border-radius: 4px
+=======
     width: 450px
     border-radius: $radius
+>>>>>>> upstream/master
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/modal/index.vue]: looks like some unresolved conflicts. Are you sure this is compiling?

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 2]
```diff
@@ -91,13 +91,32 @@
...
       position: relative
+      cursor: pointer
+    select
+      -webkit-appearance: menulist-button
+      width: 100%
+      height: 40px
+      font-weight: bold
+  .header
+    text-align: center
+  .create-btn
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue:127]: convention in our code-base is to have one blank line between classes

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 3]
```diff
@@ -91,13 +91,32 @@
...
-    input, select
+    input
       width: 100%
+      height: 40px
+      font-weight: bold
     label
       position: relative
+      cursor: pointer
+    select
+      -webkit-appearance: menulist-button
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue]: We should avoid using prefixed attributes if at all possible. Also, make sure you're testing other browsers in browserstack! ;-)

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 4]
```diff
@@ -72,7 +72,6 @@
 
   .title
     display: inline
-
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:75]: whitespace

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 5]
```diff
@@ -21,8 +21,8 @@
           <label for="username">Password</label>
           <input type="password" id="password" required v-model="user.password">
         </div>
-
         <div class="user-field">
+          <label for="role">Role</label>
           <select v-model="user.role">
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue]: good job adding the label - but you need to give the `select` dropdown `name="role"` and `id="role"` for the label to be associated with it

File: kolibri/plugins/management/assets/src/vue/user-page/index.vue [hunk 6]
```diff
@@ -213,6 +212,14 @@
...
     outline: none
 
+  .roster
+    width: 100%
+
+    /*Prevent lists that are too long*/
+    max-height: 300px
+    overflow: hidden
+    overflow-y: scroll
+
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/user-page/index.vue]: These changes don't seem related to the PR - maybe you added them by accident?

File: kolibri/plugins/management/assets/src/vue/modal/index.vue [hunk 7]
```diff
@@ -105,6 +112,7 @@
     float: right
     color: $core-text-default
     border: none
+    border-color: transparent
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/vue/modal/index.vue]: what does this line do since `border: none` is set above it?

**Comment by rayykim** [comment]
@radinamatic edit modal changes

File: kolibri/plugins/management/assets/src/vue/user-page/index.vue [hunk 8]
```diff
@@ -213,6 +212,14 @@
     height: 35px
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/index.vue:212]: It's annoying that GH is not allowing comments on folded lines... :unamused: 

@rayykim 
Anyways, this is a suggestion for line 206, for `:focus` state of the search input field. I really like inset box shadow style for input fields - [check out the one I had on KA Lite login last year](https://github.com/learningequality/ka-lite/pull/3647/files), maybe we can add it here with our own action purple color.

@DXCanas @indirectlylit: The same example contains a style for form fields with errors, whoever is working on it might implement the visual too... :wink:

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 9]
```diff
@@ -3,31 +3,32 @@
...
 
-      <h1 slot="header">Edit User</h1>
+      <h1 slot="header" class="header">Account Info</h1>
 
       <div slot="body">
 
         <div class="user-field">
-        <label for="username">Username</label>:
-        <input type="text" id="username" v-model="username_new" placeholder="Please type in your username.">
+          <label for="username">Full Name</label>:
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: `for` in `<label>` must be the same as `id` in `<input>`.

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 10]
```diff
@@ -3,31 +3,32 @@
...
+          <input type="text" class="edit-form edit-fullname" aria-label="fullname" id="name" v-model="fullName_new">
         </div>
 
         <div class="user-field">
-        <label for="password">Password</label>:
-        <input type="text" id="password" v-model="password_new" placeholder="Please type in your password.">
+          <label for="name">Username</label>:
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: Same here: `for` in `<label>` must be the same as `id` in `<input>`.

File: kolibri/plugins/management/assets/src/vue/user-page/index.vue (hunk 6 above)
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/index.vue]: I think they were fixes from the missing roster class of Tuesday's PR. It's likely they were added in because she noticed it was missing.

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 11]
```diff
@@ -57,23 +65,37 @@
...
-        this.user.facility = this.facility;
-        // using promise to ensure that the user is created before closing
-        // can use this promise to have flash an error in the modal?
-        this.createUser(this.user, this.role).then(() => {
-          for (const userProp of Object.getOwnPropertyNames(this.user)) {
-            this.user[userProp] = '';
-          }
-          this.$refs.modal.closeModal();
-        });
+        if (this.user.password && this.user.confirm_password === this.user.password) {
```
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue]: I feel like this could be streamlined significantly. I might take a look at refactoring this after MVP

File: kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue [hunk 12]
```diff
@@ -57,23 +65,37 @@
...
         role: 'learner',
       };
     },
+    computed: {
+      checkPassword() {
+        // if (this.user.password.length =< 9) {
+        //   return 'Error: too short';
+        // }
+        if (this.user.confirm_password && this.user.password !== this.user.confirm_password) {
+          return 'passwords not matched';
```
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue]: @indirectlylit How would I go about integrating this functionality with the error message stuff in https://github.com/learningequality/kolibri/pull/372?

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 13]
```diff
@@ -73,12 +98,57 @@
...
+    padding: 5px 10px
+    letter-spacing: 0.08em
+    border: none
+    border-bottom: 1px solid $core-text-default
+    height: 30px
+    &:focus
+      outline: none
+      border-bottom: 3px solid $core-action-normal
+
+  .edit-username
```
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue:140]: @rayykim any reason we've transitioned to using backgrounds rather than doing the `<svg>` tags in the html like we were doing before?

File: kolibri/plugins/management/assets/src/vue/user-page/index.vue (hunk 8 above)
Comment (DXCanas) [inline kolibri/plugins/management/assets/src/vue/user-page/index.vue:212]: I think this might be a better note for the people making the mock ups. @whitzhu @66eli77  ?

**Comment by DXCanas** [comment]
This PR is getting pretty fat. @rayykim would you mind updating the title/description with everything it does? @indirectlylit do you think it would be better to break these up in to various PR's? I was given merge access and added to the team so I can help out with some of these now.

**Comment by indirectlylit** [comment]
looks good, thanks

File: kolibri/plugins/management/assets/src/vue/user-page/index.vue (hunk 8 above)
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/index.vue:212]: @DXCanas Tota11y :cool: with me, as long as the `:focus` states are implemented. This for example looks fine to me for search input field:

```
  input[type='search']
  // snip
    &:focus
      outline: none
      border: solid 1px
      border-color: $core-bg-light
      box-shadow: 0 0 5px 1px rgba(114, 72, 111, 0.75)
```

And this would be a good for the drop-down:

```
  .user-filter-dropdown
   // snip
    &:focus
      outline: none
      box-shadow: 0 0 5px 2px rgba(114, 72, 111, 0.75) inset

  option[selected]
    background-color: $core-action-light
```

Unfortunately no way to change the default red of selected option without JS, so let's leave it at that.

**Comment by radinamatic** [comment]
A11y-wise nothing stands out as problematic, a part from missing `:focus` states for search input field and the drop-down user select. 

@DXCanas Since those are not related to the **Add New** modal, they can be implemented in another PR.

**Comment by indirectlylit** [comment]
Okay - looks like most issues have been addressed. @DXCanas want to give it a final look-over and merge?

**Comment by DXCanas** [comment]
@indirectlylit will do

**Comment by DXCanas** [comment]
Looks like password in `user-edit-modal` is missing some styles? 
image

**Comment by rayykim** [comment]
@DXCanas updating the password field right now

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 14]
```diff
@@ -2,22 +2,47 @@
 
   <div class="user-edit-modal">
     <modal btntext="Edit">
-      <div class="title" slot="header">User edit</div>
+
+      <h1 slot="header" class="header">Account Info</h1>
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: Wasn't the title "Edit User"?
If this modal opens with the "pencil" button, I believe the modal title should still include the word "edit", maybe "View and Edit Account Info"?

Comment (rayykim) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: @radinamatic in the mock-up, it said "Account Info" so it was changed. @whitzhu any thoughts?

Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: Modal offers the editing functionality so I believe the title/main heading should reflect that.

Comment (rayykim) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: @radinamatic changed it to "Edit Account Info"

File: kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue [hunk 15]
```diff
@@ -2,22 +2,47 @@
...
+        </div>
+
       </div>
+
+        <div slot="footer">
+          <button class="confirm-btn" type="button" @click="editUser">Confirm</button>
+        </div>
+
+        <button class="no-border" slot="openbtn">
+          <span class="visuallyhidden">Edit User</span>
```
Comment (radinamatic) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: Could you change the wording here to "Edit Account Info" too, since this is the button that opens the modal? :smiley:

Comment (rayykim) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: @radinamatic done :)

Comment (rayykim) [inline kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue]: screen shot 2016-08-11 at 5 17 08 pm
password feature updated;
edit title change
